### DISCUSSION TITLE
Users relationship with postgresql tables

### DISCUSSION THREAD

**Discussion by colomboe** [Budibase/budibase#3404] [Help] [answered]
Hi, I'm experimenting with BudiBase and I find it very interesting.

I started by using the internal database, where I've been able to create some tables and create some relationships with the special Users table. Now I'm switching to an external postgresql database in order to be able to execute complex SQL queries. 

But actually I immediately hit a blocker as I couldn't create relationships between the Users table (that I understand MUST be internal to BudiBase) and my postgresql external data source.

My need is to be able to show to my users only their owned records and not the full table content. And, at the same time, mark the owner of record when the user creates a new one. I'm able to to this using the internal BudiBase database, but how I can do it with postgresql?

Is there any trick that could enable me in this scenario?

**Comment by mjashanks** [comment, marked as answer]
Hi Emanuele.

This is possible, using the `{{ Current User._id }}` binding.

Whenever you save your record, you can save the `_id` of the currently logged in user, to a DB column. In the `Save Row` action (on button click), you can set the column name (I'm using the "Description" column to store the user's id, which is a bit weird):

![image](https://user-images.githubusercontent.com/3524181/142209362-8cede8fe-304f-4a8e-a338-cb35d60cddc8.png)

Then, when you are listing records, you can filter by the current user's ID:

![image](https://user-images.githubusercontent.com/3524181/142209501-61f9972f-26a8-4e0c-915e-36878342e3c6.png)

Hope this helps!

**Reply by colomboe** [reply]
Thank you, I'm going to try it ASAP. With this solution, if I need to show for a "power user" the list of all the records and the corresponding owner user (name), what should I do? Should I also save the user name in a column on the postgresql table? How? Or can I translate the user._id on the fly when showing a record in the UI?

**Reply by colomboe** [reply]
I've tried your solution and it works very well! My only problem now it how to show the user name in a table where I stored the user id. Is there a solution also for this problem?

**Reply by mjashanks** [reply]
Yea - i was trying this out and couldn't see a good way of doing it. Your options at the moment are
1. Copy the user name - like you say
2. Create a table of users in your SQL database, and setup a relationship.

I logged this issue as a possible future solution for this: https://github.com/Budibase/budibase/issues/3420

**Reply by colomboe** [reply]
Thank you! About the solution 2, is there an easy way to see the _id of the users from the BudiBase users administration interface? In order to be able to populate the table on the SQL database.

**Reply by mjashanks** [reply]
Good point! If you go to your Users table and click "Show Auto Columns", you can copy the `_id` to your clipboard....

![image](https://user-images.githubusercontent.com/3524181/142391304-4814b5dc-227d-4026-951f-97d4f232e9ff.png)

**Reply by colomboe** [reply]
I don't know why, but I can't see all the users of my app in that page... Could it depend of the role of the user?

**Reply by colomboe** [reply]
I've found out this: https://github.com/Budibase/budibase/discussions/3329

Given this, I have all my answers 😄 

I'll keep an eye on #3420

Thank you very much for your help!
